reference/forms/types/dateinterval.rst

```diff
@@ -0,0 +1,325 @@
+|                      | - `widget`_                                                                       |
+|                      | - `with_days`_                                                                    |
+|                      | - `with_hours`_                                                                   |
+|                      | - `with_invert`_                                                                  |
+|                      | - `with_minutes`_                                                                 |
+|                      | - `with_months`_                                                                  |
+|                      | - `with_seconds`_                                                                 |
+|                      | - `with_weeks`_                                                                   |
+|                      | - `with_years`_                                                                   |
+|                      | - `years`_                                                                        |
++----------------------+-----------------------------------------------------------------------------------+
+| Inherited            | - `data`_                                                                         |
+| options              | - `disabled`_                                                                     |
+|                      | - `inherit_data`_                                                                 |
+|                      | - `invalid_message`_                                                              |
+|                      | - `invalid_message_parameters`_                                                   |
+|                      | - `mapped`_                                                                       |
++----------------------+-----------------------------------------------------------------------------------+
+| Parent type          | :doc:`form </reference/forms/types/form>`                                         |
++----------------------+-----------------------------------------------------------------------------------+
+| Class                | :class:`Symfony\\Component\\Form\\Extension\\Core\\Type\\DateIntervalType`        |
++----------------------+-----------------------------------------------------------------------------------+
+
+Basic Usage
+-----------
+
+This field type is highly configurable, but easy to use. The most important
+options are ``input`` and ``widget``.
+
+Suppose that you have a ``remindEvery`` field whose underlying interval is a
+``DateInterval`` object. The following configures the ``dateinterval`` type
+for that field as **three different choice fields**:
+
+.. code-block:: php
+
+    $builder->add('remindEvery', DateIntervalType::class, array(
+        'widget' => 'choice',
```

Comment:
Should be `ChoiceType::class`, correct? And we usually add the `use` statements above :)

```rst
 use Symfony\Component\Form\Extension\Core\Type\DateIntervalType;
 // ...

$builder...
```